src/space_stage/FleetMovementOrder.cs

```diff
@@ -18,22 +18,46 @@ public FleetMovementOrder(SpaceFleet fleet, Vector3 targetPosition) : base(fleet
     protected override bool WorkOnOrder(float delta)
     {
         var unitPosition = Unit.GlobalPosition;
+
         var toTarget = TargetPosition - unitPosition;
 
         var distanceToTarget = toTarget.Length();
 
         var adjustedSpeed = delta * Unit.Speed;
 
-        // TODO: adjust the fleet rotation towards the travel direction
+        // The normalized vector from the ship to the target point
+        var direction = toTarget / distanceToTarget;
+
+        var currentRotation = new Quaternion(new Vector3(0, 0, 1), Unit.Transform.Basis.Z.Normalized()).Normalized();
+        var targetRotation = new Quaternion(new Vector3(0, 0, 1), toTarget.Normalized()).Normalized();
 
-        if (distanceToTarget < adjustedSpeed)
+        if (direction.Dot(Unit.Transform.Basis.Z) < 0.99f)
         {
-            Unit.GlobalPosition = TargetPosition;
-            return true;
+            var smoothRotation = currentRotation.Slerp(targetRotation, 0.8f * delta);
+            Unit.Transform = new Transform3D(new Basis(smoothRotation), unitPosition);
+
+            Unit.Transform = Unit.Transform.Orthonormalized();
```

Comment:
Why does this use `Orthonormalized` here? I've never used that in Godot.

Reply:
That was based on a guide that recommended to do that when rotating - but actually the new transform would achieve the same thing anyway. I'll remove it.